Rewards summary message on game end

reussy opened the pull request:
This implements the summary message and closes #206

 **Some methods are moved by Intellij reformat code tool!**
- New maps for keep the current coins and experience earned until game end. It's unnecessary create classes for this (I guess).
- These maps are loaded on player join and destroyed on player leave.
- These maps are updated on `addXp` method and `MoneyListeners` methods.
- The message is implemented below of top killers message.
- The radical implements of "long progress bar" is for use in chat. The shorter progress bar it looks very ugly in the rewards summary message. **This could be improved or customized from level.yml in the future.**

Everything was tested by me. It can be tested by you at `mc.exoduscraft.club - BedWars Multiarena`
Looks like this: https://i.imgur.com/rDPjXkw.png

Commit:
Rewards summary message on game end

This implements the summary message and closes #

* Some methods are moved by Intellij reformat code tool!
- New maps for keep the current coins and experience earned until. game end. It's unnecessary create classes for this (I guess).
- These maps are loaded on player join and destroyed on player leave.
- These maps are updated on addXp method and MoneyListeners methods.
- The message is implemented below of top killers message.
- The radical implements of "long progress bar" is for use in chat. The shorter progress bar it looks very ugly in the rewards summary message. This could be improved or customized from level.yml in the future.

Everything was tested by me. It can be tested by you at mc.exoduscraft.club - BedWars Multiarena
Looks like this: https://i.imgur.com/rDPjXkw.png

Commit:
Rewards summary message on game end

This implements the summary message and closes #

* Some methods are moved by Intellij reformat code tool!
- New maps for keep the current coins and experience earned until. game end. It's unnecessary create classes for this (I guess).
- These maps are loaded on player join and destroyed on player leave.
- These maps are updated on addXp method and MoneyListeners methods.
- The message is implemented below of top killers message.
- The radical implements of "long progress bar" is for use in chat. The shorter progress bar it looks very ugly in the rewards summary message. This could be improved or customized from level.yml in the future.

Everything was tested by me. It can be tested by you at mc.exoduscraft.club - BedWars Multiarena
Looks like this: https://i.imgur.com/rDPjXkw.png

Commit:
Minimal changes

Commit:
Minimal changes

Commit:
Reward summary message path for other languages added

Commit:
Reward summary message path for other languages added

Commit:
Fixed placeholder with incorrect name

Commit:
Fixed placeholder with incorrect name

ajgeiss0702 committed:
Warnings changes (#264)

* Add seperate warnings listener, make warnings delayed, add spawn-protection warning

* Change colors in multiverse messages

andrei1058 commented on bedwars-plugin/src/main/java/com/andrei1058/bedwars/levels/internal/PlayerLevel.java:
```diff
@@ -33,21 +34,19 @@
 public class PlayerLevel {
 
-    private UUID uuid;
+    private static final ConcurrentHashMap<UUID, PlayerLevel> levelByPlayer = new ConcurrentHashMap<>();
```
why is this concurrent?

reussy replied:
I think that's was touched by "reformat code" tool by Intellij. In the current code it is concurrent. I must have missed it when sending the pr

![image](https://user-images.githubusercontent.com/63428864/152575589-e7a50ae1-27bb-407a-9498-e48323b270b6.png)

andrei1058 commented on bedwars-plugin/src/main/java/com/andrei1058/bedwars/arena/Arena.java:
```diff
@@ -158,6 +159,9 @@
     private HashMap<Player, Integer> playerFinalKillDeaths = new HashMap<>();
 
+    /* SUMMARY REWARDS */
+    private static final Map<UUID, Integer> coinsEarned = new HashMap<>();
```
it should not be static

andrei1058 commented on bedwars-plugin/src/main/java/com/andrei1058/bedwars/levels/internal/PlayerLevel.java:
```diff
@@ -174,8 +210,12 @@ public void addXp(int xp, PlayerXpGainEvent.XpSource source) {
         Bukkit.getPluginManager().callEvent(new PlayerXpGainEvent(Bukkit.getPlayer(uuid), xp, source));
         modified = true;
+        if (Arena.getExperienceEarned().containsKey(uuid)) {
+            Arena.getExperienceEarned().put(uuid, Arena.getExperienceEarned(uuid) + xp);
```
this must be handled with an event listener XpGainEvent etc.

andrei1058 commented on bedwars-plugin/src/main/java/com/andrei1058/bedwars/money/internal/MoneyPerMinuteTask.java:
```diff
@@ -26,6 +26,7 @@ public MoneyPerMinuteTask(Arena arena) {
             for (Player p : arena.getPlayers()) {
                     BedWars.getEconomy ().giveMoney ( p, money );
                     p.sendMessage ( Language.getMsg ( p, Messages.MONEY_REWARD_PER_MINUTE ).replace ( "{money}", String.valueOf ( money ) ) );
+                    Arena.getCoinsEarned().put(p.getUniqueId(), Arena.getCoinsEarned(p.getUniqueId()) + money);
```
we should implement a new event like BedWarsMoneyGainEvent and listen to that in otder to track coins earned

PikaMug committed:
Make arena accessible from shop events (#222)

andrei1058 committed:
fix #178 (#232)

Itsmemonzu committed:
Hindi Language (#57)

* Created Hindi language

* Added Hindi in BedWars,java

* Deleted Hindi Language

* Re-added Hindi Language

* Fixed Hindi.java

* Fixed a few mistakes in Hindi.java

Translated a missed word for RESPAWN_TITLE in Hindi.java
Added translation for MONEY_REWARD

* Fixed conflict

* Fixed conflict

* Translated a missed sentence in Hindi.java

danirod12 committed:
Update Russian language lines (#79)

* Update Russian translation, added missing lines

Updated Russian translation, added missing translations

* Update Russian translation, added missing lines

andrei1058 committed:
add possibility to toggle player sidebar via API

andrei1058 committed:
version string update

andrei1058 committed:
version name fix

danirod12 committed:
Added LuckyBlocks community addon to README.md (#273)

Last commit was canceled because i not a contributor. Now i am already a contributor.

Added LuckyBlocks NTD community addon

FabrisNick committed:
Fix registering datawatcher object after entity initialization (#305)

Co-authored-by: BuildTools <unconfigured@null.spigotmc.org>

danirod12 committed:
Make UpgradeBuyEvent Cancellable (#313)

* Make UpgradeBuyEvent cancellable

palash-08 committed:
Update README.md (#324)

andrei1058 committed:
fix player skull crashing the server (#233)

* fix #210

* hotfix for #223 and #210

andrei1058 committed:
fix message placeholder replacement and chat clean up (#231)

fix message placeholder replacement and chat clean up (#231)

ajgeiss0702 committed:
add a space after next page (and fix the color) (#332)

ajgeiss0702 committed:
skipped-slot item improvements (#330)

* allow disabling of skipped-slot items

* allow completly changing name and lore of skipped-items item

* disable skipped-items item if material is air

reussy committed:
Better particles for SpoilTNT feature (#270)

* SpoilTNT Particle improved

Makes player particle with TNT more similar to Hypixel

* Changed from 5 ticks to 20 ticks

Co-authored-by: BuildTools <unconfigured@null.spigotmc.org>

ajgeiss0702 committed:
Dont allow opening of shops after game ends (#334)

* prevent players from opening shops after game ends

* close any open guis for arena players when the game ends

andrei1058 commented:
bump

ajgeiss0702 committed:
Fix fireballs on 1.12 (#241)

* fireball knockback WIP

* move fireball listeners to a seperate class

* fix lighting blocks next to bed on fire with fireball

* cancel fireball direct hits

lots of damage

* fine-tune fireball values

* add fireball cooldown

* whoops, forgot to remove this

* Add LastHit

* Customisable damage for self, team, and enemies

* Change default cooldown to 0.5

* dont use scheduler in fireball cooldown

* switch to ProjectileHitEvent instead of EntityExplodeEvent

* remove unnecisary code

* FINALLY really fix fireballs on <1.12

ajgeiss0702 committed:
make firetick disabled by default (#360)

ajgeiss0702 committed:
add {vSuffix} to join and leave messages (#361)

ajgeiss0702 committed:
fix fireballs making fire even if make-fire is disabled (#359)

ajgeiss0702 committed:
fix group name in arena gui not coming from the language file (#354)

ajgeiss0702 committed:
add setRespawnTime to PlayerReJoinEvent (#353)

ghost commented:
When will this be implemented

reussy committed:
Merge remote-tracking branch 'origin/reward-summary-message' into reward-summary-message

reussy committed:
Well, I have made use of the PlayerXpGainEvent event to update the experience gained in the course of the game. On the other hand a new event is created "PlayerMoneyGainEvent" that fulfills the same function, to update the money earned. This event is called in each method that money is given to the player.

Is there a better way to store the experience and money earned instead of a static hashmap?

Also remove odd spaces, improve some variable names to follow Java conventions and make the code more homogeneous with respect to other variables.

reussy commented:
Well, I don't know what the hell happened but well, I'd better make a new pr :/
